src/main/scala/org/ergoplatform/ErgoAddress.scala

```diff
@@ -203,6 +205,17 @@ case class ErgoAddressEncoder(networkPrefix: Byte) {
       }
     }
   }
+
+  def fromPropositionBytes(bytes: Array[Byte]): Try[ErgoAddress] = Try {
+    val script: Value[SType] = ValueSerializer.deserialize(bytes)
+    bytes.head match {
```

Comment:
@oskin1 Did you tried matching on the deserialized 'script' itself? It should work here. You should get AST nodes `ProveDlog`, `AND`, `Value[SBoolean.type]`.

Reply:
@greenhat, yes i did, but in case of P2SH we need to match by mask, which contains numerous opcodes - it would be inconvenient to match such structure (AST) in deserialised form

Reply:
@oskin1 I see matching by `OpCodes.AndCode`. I've tried: 
```
  def fromPropositionBytes(bytes: Array[Byte]): Try[ErgoAddress] = Try {
    ValueSerializer.deserialize(bytes) match {
      case d @ ProveDlog(_) =>
        P2PKAddress(d)
      case a @ AND(_) =>
        Pay2SHAddress(a)
      case b: Value[SBoolean.type] @unchecked if b.tpe == SBoolean =>
        Pay2SAddress(b)
      case _ => throw new RuntimeException()
    }
```
and the test is still green. Am I missing something?

Reply:
@oskin1 Excellent! Thanks! Only minor note with test location left.